src/pages/index.js

```diff
@@ -54,39 +72,136 @@ const profileEditModal = new ModalWithForm(
   handleProfileEditSubmit
 );
 profileEditModal.setEventListeners();
+
 const addCardModal = new ModalWithForm("#add-card-modal", handleAddCardCreate);
 addCardModal.setEventListeners();
 
+const profileAvatarEditModal = new ModalWithForm(
+  "#profile-avatar-edit-modal",
+  handleProfileAvatarUpdate
+);
+profileAvatarEditModal.setEventListeners();
+
+// Add an event listener to the button to open the avatar update modal
+const profileAvatarEditButton = document.querySelector(
+  "#profile-avatar-edit-button"
+);
+profileAvatarEditButton.addEventListener("click", () => {
+  profileAvatarEditModal.open();
+  profileAvatarInput.value = "";
+});
+
 /*    Form validators    */
-console.log(document.forms);
 const profileEditForm = document.forms["profile-edit-form"];
 const addCardForm = document.forms["add-card-form"];
+const profileAvatarForm = document.forms["profile-avatar-edit-form"];
 const editFormValidator = new FormValidator(config, profileEditForm);
 const addFormValidator = new FormValidator(config, addCardForm);
+const avatarFormValidator = new FormValidator(config, profileAvatarForm);
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarFormValidator.enableValidation();
 
 /*    ModalWithImage Instances    */
 const previewImageModal = new ModalWithImage("#preview-image-modal");
 previewImageModal.setEventListeners();
 
+// Handle profile avatar update function
+function handleProfileAvatarUpdate(data) {
+  api
```

Comment:
According to the task you need to change submit buttons text (`Saving…`) while waiting for the server response. And the user should see it.